src/user/TidewiseMessages.c

```diff
@@ -57,6 +57,10 @@ BOOL Fill_e4PacketPayload(uint8_t *payload, uint8_t *payloadLen) {
         pld->pos[i] = dData[i];
     }
 
+    real geoid2ellipsoid;
+    EKF_GetGeoidAboveEllipsoid(&geoid2ellipsoid);
```

Comment:
is GeoidAboveEllipsoid the same as AboveMSL?

Reply:
Ellipsoid == Ellipse that is approximating the overall shape of the earth
Geoid == MSL

How I remember is that ellipse is ... an ellipse, i.e. a regular shape, and that therefore the geoid is the "lumpy" one that takes irregularities into account.

Which is why GeoidAboveEllipsoid is geoid2ellipsoid (as per the convention in https://www.rock-robotics.org/rock-and-syskit/libraries/geometric_transformations.html). And since `pos` is `imu2ellipsoid` we have

`imu2msl = imu2geoid = ellipsoid2geoid + imu2ellipsoid = -geoid2ellipsoid + imu2ellipsoid;`

Reply:
good explanation, got it!